trick_source/sim_services/MonteCarlo/MonteVarFile.cpp

```diff
@@ -30,64 +30,78 @@ std::string Trick::MonteVarFile::describe_variable()
     std::stringstream ss;
 
     ss << "#NAME:\t\t" << this->name << "\n"
-       << "#TYPE:\t\tFILE\n" 
+       << "#TYPE:\t\tFILE\n"
        << "#UNIT:\t\t" << this->unit << "\n"
        << "#FILE:\t\t" << this->file_name << "\n"
        << "#COLUMN:\t" << this->column << "\n";
 
     return ss.str();
 }
 
-std::string Trick::MonteVarFile::get_next_value() {
-    if (input_file_stream->good()) {
-        double file_value;
-        char *current_position = buffer;
-        do {
+std::string Trick::MonteVarFile::get_next_value()
+{
+    if (input_file_stream->good())
+    {
+        // Skip the comments and empty lines in the data file.
+        do
+        {
             input_file_stream->getline(buffer, 4096);
-            file_value = strtold(buffer, &current_position);
-            if (input_file_stream->eof()) {
+
+            if(input_file_stream->eof())
+            {
                 input_file_stream->close();
                 return "EOF";
             }
-        } while (file_value == 0 && current_position == buffer && input_file_stream->good());
+        }
+        while(buffer[0] == '#' || buffer[0] == '\0');
```

Comment:
This is a change to the "comment skipping" logic. Previously, any line that didn't start with a number as determined by `strtold` would be skipped. The new logic is more restrictive, skipping only empty lines and lines that start with `#`. This will break people who rely on the old behavior. Such is the cost of progress!

Reply:
This is correct, however I can not think of a way of skipping comments whilst simultaneously accepting strings as possible input values. The old functionality totally disregarded any value that could not be parsed as a long double. Accepting a string requires a way of skipping comments and null lines.

In the tutorial and user guide, the suggest way of specifying comment lines was to use a '#' character. This change is consistent with that documentation.